buildvertical.targets

```diff
@@ -138,16 +142,34 @@
     <PropertyGroup>
       <_SelectedTargetGroup>%(VerticalTargets.TargetGroup)</_SelectedTargetGroup>
       <_SelectedOSGroup>%(VerticalTargets.OSGroup)</_SelectedOSGroup>
+      <_Configuration>$(_SelectedTargetGroup)</_Configuration>
+      <_Configuration Condition="'$(_SelectedOSGroup)' != '' and '$(_SelectedOSGroup)' != 'AnyOS'">$(_Configuration)-$(_SelectedOSGroup)</_Configuration>
+      <_AdditionalProperties Condition="'$(_SelectedTargetGroup)' != ''">TargetGroup=$(_SelectedTargetGroup)</_AdditionalProperties>
+      <_AdditionalProperties Condition="'$(_SelectedOSGroup)' != '' and '$(_SelectedOSGroup)' != 'AnyOS'">OSGroup=$(_SelectedOSGroup);$(_AdditionalProperties)</_AdditionalProperties>
+      <_AdditionalProperties Condition="'$(_Configuration)' != ''">Configuration=$(_Configuration);$(_AdditionalProperties)</_AdditionalProperties>
     </PropertyGroup>
-    <Message Importance="Low" Text="Annotate ProjectReference - $(MSBuildProjectFullPath) - $(_ProjectReferenceToAnnotate) ($(_SelectedTargetGroup) === $(_DefaultTargetGroup),  $(_SelectedOSGroup)) Supports: $(BuildConfigurations)" />
+    <Message Importance="Normal" Text="Annotate ProjectReference - $(MSBuildProjectFullPath) - $(_ProjectReferenceToAnnotate) ($(_SelectedTargetGroup),  $(_SelectedOSGroup)) Supports: $(BuildConfigurations)" />
 
     <Error ContinueOnError="true" Condition="$(_ProjectReferenceToAnnotate.EndsWith('.builds')) or $(_ProjectReferenceToAnnotate.EndsWith('.pkgproj'))" Text="Warning: $(MSBuildProjectFullPath) contains ProjectReference to a multi-build project, '$(_ProjectReferenceToAnnotate)'" />
     <!-- Annotate the ProjectReference -->
     <ItemGroup>
-        <ProjectReference Condition="'%(ProjectReference.OutputItemType)' != 'ResolvedMatchingContract'">
-          <TargetGroup Condition="'%(ProjectReference.TargetGroup)' == ''">$(_SelectedTargetGroup)</TargetGroup>
-          <OSGroup Condition="'%(ProjectReference.OSGroup)' == ''">$(_SelectedOSGroup)</OSGroup>
+        <ProjectReference Condition="'%(ProjectReference.Identity)' == '$(_ProjectReferenceToAnnotate)' and '$(BuildConfigurations)' != ''">
+          <TargetGroup>$(_SelectedTargetGroup)</TargetGroup>
+          <OSGroup>$(_SelectedOSGroup)</OSGroup>
+          <Configuration>$(_Configuration)</Configuration>
+          <AdditionalProperties>$(_AdditionalProperties);%(ProjectReference.AdditionalProperties)</AdditionalProperties>
         </ProjectReference>
+
+        <ProjectReference Condition="'%(ProjectReference.Identity)' == '$(_ProjectReferenceToAnnotate)' and ('$(_SelectedOSGroup)'  == '' or '$(_SelectedOSGroup)' == 'AnyOS')">
+          <UndefineProperties>OSGroup;%(ProjectReference.UndefineProperties)</UndefineProperties>
```

Comment:
We shouldn't need to worry about undefining OSGroup/TargetGroup/Configuration group because I'm thinking we will eventually forcibly override and ignore them and always set them from Configuration. However I'm fine keeping this for now if it unblock your work.